src/ctelnet.cpp

```diff
@@ -1908,3 +1911,39 @@ void cTelnet::raiseProtocolEvent(const QString& name, const QString& protocol)
     event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
     mpHost->raiseEvent(event);
 }
+
+// credit: https://github.com/qflow/websockets
+void cTelnet::setKeepAlive(int socketHandle)
+{
+    constexpr int timeout = 2 * 60; //  /* send keepalive after 2 minutes */
+#if defined(Q_OS_WIN32)
+    struct tcp_keepalive
+    {
+        u_long onoff;
+        u_long keepalivetime;
+        u_long keepaliveinterval;
+    } alive;
+    alive.onoff = TRUE;
+    alive.keepalivetime = timeout * 1000;
+    alive.keepaliveinterval = 3000;
+    DWORD dwBytesRet = 0;
+    int res = WSAIoctl(socketHandle, SIO_KEEPALIVE_VALS, &alive, sizeof(alive), NULL, 0, &dwBytesRet, NULL, NULL);
+
+#elif defined(Q_OS_LINUX)
+    int enableKeepAlive = 1;
+    int res = setsockopt(socketHandle, SOL_SOCKET, SO_KEEPALIVE, &enableKeepAlive, sizeof(enableKeepAlive));
+
+    res = setsockopt(socketHandle, IPPROTO_TCP, TCP_KEEPIDLE, &timeout, sizeof(timeout));
+
+    int count = 3; // send up to 3 keepalive packets out, then disconnect if no response
+    res = setsockopt(socketHandle, SOL_TCP, TCP_KEEPCNT, &count, sizeof(count));
+
+    int interval = 2; // send a keepalive packet out every 2 seconds (after the 5 second idle period)
+    res = setsockopt(socketHandle, SOL_TCP, TCP_KEEPINTVL, &interval, sizeof(interval));
+
+#elif defined(Q_OS_MACOS)
+    constexpr int on = 1;
+    setsockopt(socketHandle, SOL_SOCKET,  SO_KEEPALIVE, &on, sizeof(on));
```

Comment:
If you compare the Linux version to the macOs version you may see that this line and the one above are functionally the same for both OSes; and `res` is not actually used for either Windows or Linux so there is no need to have it to capture the returned value from the function calls (unless wanted for debugging)...

Reply:
> If you compare the Linux version to the macOs version you may see that this line and the one above are functionally the same for both OSes

I prefer to keep the code this way instead of making it more complicated with the "if windows, if macos and linux, if linux only" sort of way. It's simpler to understand: "if windows, if linux, if macos".

Removed `res`.